parachain/pallets/inbound-queue/src/lib.rs

```diff
@@ -276,15 +275,32 @@ pub mod pallet {
 	}
 
 	impl<T: Config> Pallet<T> {
-		pub fn do_convert(message: inbound::VersionedMessage) -> Result<Xcm<()>, Error<T>> {
-			let xcm =
-				T::MessageConverter::convert(message).map_err(|e| Error::<T>::ConvertMessage(e))?;
+		pub fn do_convert(
+			message: inbound::VersionedMessage,
+			dest: MultiLocation,
+		) -> Result<Xcm<T::RuntimeCall>, Error<T>> {
+			let xcm = T::MessageConverter::convert(message, dest)
+				.map_err(|e| Error::<T>::ConvertMessage(e))?;
 			Ok(xcm)
 		}
 
-		pub fn send_xcm(xcm: Xcm<()>, dest: ParaId) -> Result<XcmHash, Error<T>> {
-			let dest = MultiLocation { parents: 1, interior: X1(Parachain(dest.into())) };
-			let (xcm_hash, _) = send_xcm::<T::XcmSender>(dest, xcm).map_err(Error::<T>::from)?;
+		pub fn do_execute(
+			origin: MultiLocation,
+			message: Xcm<T::RuntimeCall>,
+		) -> Result<XcmHash, Error<T>> {
+			let mut xcm_hash = message.using_encoded(sp_io::hashing::blake2_256);
+
+			// Attempt to send XCM to a dest parachain
+			T::XcmExecutor::prepare_and_execute(
+				origin,
+				message,
+				&mut xcm_hash,
+				Weight::zero(),
+				Weight::zero(),
+			)
+			.ensure_complete()
+			.map_err(Error::<T>::from)?;
+
```

Comment:
I don't think its necessary to run XCM locally. Seems like this adds unnecessary complexity.

Instead burn the fee amount from AssetHub's sovereign account, using [T::Token::burn_from](https://github.com/Snowfork/polkadot-sdk/blob/f7fbba753247de67d36af870b2c69b8f2efc629a/substrate/frame/support/src/traits/tokens/fungible/regular.rs#L252) and then include a [ReceiveTeleportedAsset](https://github.com/paritytech/polkadot/blob/962bc21352f5f80a580db5a28d05154ede4a9f86/xcm/src/v3/mod.rs#L406) instruction in the message you send to AssetHub. That will send the fee amount to the holding there.